code/game/objects/structures/railings.dm

```diff
@@ -50,13 +50,11 @@
 		return TRUE
 
 /obj/structure/railing/deconstruct(disassembled)
-	. = ..()
-	if(!loc) //quick check if it's qdeleted already.
-		return
 	if(!(flags_1 & NODECONSTRUCT_1))
```

Comment:
You need to change this to not allow deconstruct, otherwise this just does the same bug where it doesn't drop rods.

Reply:
I'm not sure what you mean here, it worked in-game when I tested it

Reply:
For things with NODECONSTRUCT_1?

Reply:
Oh nevermind, I was thinking of a different flag.